django_irods/views.py

```diff
@@ -126,17 +128,26 @@ def download(request, path, rest_call=False, use_async=True, use_reverse_proxy=T
             if __debug__:
                 logger.debug("request for single file {}".format(path))
             is_sf_request = True
+
+            # check for single file aggregations
+            if "data/contents/" in path:  # not a metadata file
+                for f in ResourceFile.objects.filter(object_id=res.id):
+                    if path == f.storage_path:
+                        is_sf_agg_file = True
+                        if not is_zip_request and f.has_logical_file and \
+                                f.logical_file.is_single_file_aggregation:
+                            if 'url' in f.logical_file.extra_data:
+                                download_url = request.GET.get('url_download', 'false').lower()
+                                if download_url == 'false':
+                                    # redirect to referenced url in the url file instead
```

Comment:
I hate to add complexity to this very complex method because we eventually need to break it up.  I wonder if you could wrap this download method and do the check for ResourceFile.logical_file.extra_data, which redirects to the url if it exists, or calls this original download method if it does not.  I don't like this functionality hidden in here.